File: samples/server/petstore/nodejs/service/UserService.js
'use strict';


/**
 * Create user
 * This can only be done by the logged in user.
 *
 * user User Created user object
 * no response value expected for this operation
 **/
exports.createUser = function(user) {
  return new Promise(function(resolve, reject) {
    resolve();
  });
}


/**
 * Creates list of users with given input array
 *
 * user List List of user object
 * no response value expected for this operation
 **/
exports.createUsersWithArrayInput = function(user) {
  return new Promise(function(resolve, reject) {
    resolve();
  });
}


/**
 * Creates list of users with given input array
 *
 * user List List of user object
 * no response value expected for this operation
 **/
exports.createUsersWithListInput = function(user) {
  return new Promise(function(resolve, reject) {
    resolve();
  });
}


/**
 * Delete user
 * This can only be done by the logged in user.
 *
 * username String The name that needs to be deleted
 * no response value expected for this operation
 **/
exports.deleteUser = function(username) {
  return new Promise(function(resolve, reject) {
    resolve();
  });
}


/**
 * Get user by user name
 *
 * username String The name that needs to be fetched. Use user1 for testing.
 * returns User
 **/
exports.getUserByName = function(username) {
  return new Promise(function(resolve, reject) {
    var examples = {};
    examples['application/json'] = {
  "firstName" : "firstName",
  "lastName" : "lastName",
  "password" : "password",
  "userStatus" : 6,
  "phone" : "phone",
  "id" : 0,
  "email" : "email",
  "username" : "username"
};
    if (Object.keys(examples).length > 0) {
      resolve(examples[Object.keys(examples)[0]]);
    } else {
      resolve();
    }
  });
}


/**
 * Logs user into the system
 *
 * username String The user name for login
 * password String The password for login in clear text
 * returns String
 **/
exports.loginUser = function(username,password) {
  return new Promise(function(resolve, reject) {
    var examples = {};
    if (Object.keys(examples).length > 0) {
      resolve(examples[Object.keys(examples)[0]]);
    } else {
      resolve();
    }
  });
}


/**
 * Logs out current logged in user session
 *
 * no response value expected for this operation
 **/
exports.logoutUser = function() {
  return new Promise(function(resolve, reject) {
    resolve();
  });
}


/**
 * Updated user
 * This can only be done by the logged in user.
 *
 * username String name that need to be deleted
 * user User Updated user object
 * no response value expected for this operation
 **/
exports.updateUser = function(username,user) {
  return new Promise(function(resolve, reject) {
    resolve();
  });
}

File: samples/server/petstore/java-dubbo/src/main/java/org/openapitools/example/api/interfaces/UserService.java
package org.openapitools.example.api.interfaces;

import java.time.OffsetDateTime;
import org.openapitools.example.model.User;
import org.openapitools.example.model.*;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;
import java.time.LocalDate;
import java.time.LocalDateTime;
import javax.annotation.Generated;


@Generated(value = "org.openapitools.codegen.languages.JavaDubboServerCodegen", comments = "Generator version: 7.17.0-SNAPSHOT")

public interface UserService {

    /**
     * Create user
     * This can only be done by the logged in user.
     *
     * @param user Created user object (required)
     * @return void
     */
    void createUser(
        User user
    );

    /**
     * Creates list of users with given input array
     * 
     *
     * @param user List of user object (required)
     * @return void
     */
    void createUsersWithArrayInput(
        List<User> user
    );

    /**
     * Creates list of users with given input array
     * 
     *
     * @param user List of user object (required)
     * @return void
     */
    void createUsersWithListInput(
        List<User> user
    );

    /**
     * Delete user
     * This can only be done by the logged in user.
     *
     * @param username The name that needs to be deleted (required)
     * @return void
     */
    void deleteUser(
        String username
    );

    /**
     * Get user by user name
     * 
     *
     * @param username The name that needs to be fetched. Use user1 for testing. (required)
     * @return User
     */
    User getUserByName(
        String username
    );

    /**
     * Logs user into the system
     * 
     *
     * @param username The user name for login (required)
     * @param password The password for login in clear text (required)
     * @return String
     */
    String loginUser(
        String username,
        String password
    );

    /**
     * Logs out current logged in user session
     * 
     *
     * @return void
     */
    void logoutUser(
    );

    /**
     * Updated user
     * This can only be done by the logged in user.
     *
     * @param username name that need to be deleted (required)
     * @param user Updated user object (required)
     * @return void
     */
    void updateUser(
        String username,
        User user
    );
}
